pykrige/ok.py

```diff
@@ -741,13 +739,36 @@ def execute(self, style, xpoints, ypoints, mask=None, backend='vectorized',
             c_pars = {key: getattr(self, key) for key in ['Z', 'eps', 'variogram_model_parameters', 'variogram_function']}
 
         if n_closest_points is not None:
+            if self.coordinates_type == 'geographic':
+                # To make use of the KDTree, we have to convert the
+                # spherical coordinates into three dimensional Euclidean
+                # coordinates, since the standard KDTree cannot handle
+                # the periodicity.
+                # Do the conversion just for the step involving the KDTree:

```

Comment:
Are we sure the k-NN in Euclidean space, will produce the same neighbors as when using the great_circle distance directly? I guess they are..

Reply:
If we discard numerical errors, it should, since there is that monotonic relation between the great circle and the 3d Euclidean distance (which was also used in the ```euclid3d_to_great_circle```).

Now with finite numerical precision I'm sure it's possible to construct examples at where the two differ. I would estimate that that happens at close distances or in case distances do not differ much. I guess that those are also configurations where one would have to be careful about the choice of NN in any case.

I think a useful thing could be to determine (k+1) NN, compare the distance difference between k and k+1, and warn if they are too close - that may be useful for Euclidean as well. Maybe in another PR?